File: src/main/java/me/deecaad/weaponmechanicsplus/weapon/modifiers/util/Whitelist.java
package me.deecaad.weaponmechanicsplus.weapon.modifiers.util;

import java.util.Collection;

public class Whitelist<E> {

    private boolean isWhitelist;
    private Collection<E> list;

    public Whitelist(boolean isWhitelist, Collection<E> list) {
        this.isWhitelist = isWhitelist;
        this.list = list;
    }

    public boolean isWhitelist() {
        return isWhitelist;
    }

    public void setWhitelist(boolean whitelist) {
        isWhitelist = whitelist;
    }

    public Collection<E> getList() {
        return list;
    }

    public void setList(Collection<E> list) {
        this.list = list;
    }

    public boolean isWhitelisted(E object) {
        return isWhitelist == list.contains(object);
    }
}

File: src/main/kotlin/com/cjcrafter/weaponmechanicsplus/weapon/modifiers/util/Whitelist.kt
/*
 * Copyright (c) 2026. All rights reserved. Distribution of this file, similar
 * files, related files, or related projects is strictly controlled.
 */

package com.cjcrafter.weaponmechanicsplus.weapon.modifiers.util

class Whitelist<E>(private var isWhitelist: Boolean, private var list: Collection<E>) {

    fun isWhitelisted(`object`: E): Boolean {
        return isWhitelist == list.contains(`object`)
    }
}